libass/ass_fontselect.c

```diff
@@ -405,10 +405,11 @@ ass_font_provider_add_font(ASS_FontProvider *provider,
                 meta->postscript_name : meta->extended_family;
             face = ass_face_stream(selector->library, selector->ftlibrary,
                                    name, &stream, index);
-        } else {
+        } else if (path) {
             face = ass_face_open(selector->library, selector->ftlibrary,
                                  path, meta->postscript_name, index);
-        }
+        } else
+            goto error;
```

Comment:
Ah, you're completely right, it was my C++ habits getting in place. After all FT_Face is just a pointer in the end (`typedef struct FT_FaceRec_*  FT_Face;`), so it can be just initialized to NULL (if that was a C++ object with constructor, then that wouldn't work out)